### PR TITLE
Added `Rename ride` function to address issue #125

### PR DESCRIPTION
**Pull request by Manticore-007**
I added the possibilty to rename a ride so you are able to give the ride a longer name than ingame. This can be benefitial in multiplayer servers to add textformat, like colour or font size to the names. (see issue #125)

image

### CODE REVIEW

File: package-lock.json [hunk 1]
```diff
@@ -2,6 +2,7 @@
 	"name": "openrct2-ride-vehicle-editor",
 	"version": "1.0.0",
 	"lockfileVersion": 3,
+	"lockfileVersion": 3,
```
Comment (Basssiiie) [inline package-lock.json:5]: This file shouldn't be part of the PR, could you revert these changes?

File: src/ui/mainWindow.ts [hunk 2]
```diff
@@ -37,7 +37,7 @@ model._selectedRide.subscribe(r =>
 
 export const mainWindow = window({
 	title,
-	width: { value: 500, min: 465, max: 560 },
+	width: 500, minWidth: 465, maxWidth: 560,
```
Comment (Basssiiie) [inline src/ui/mainWindow.ts:40]: It seems you are reverting some changes from a more recent version here? This file should probably also not be in this PR as it doesn't add anything related to the renaming functionality.

File: src/ui/rideWindow.ts [hunk 3]
```diff
@@ -21,10 +22,10 @@ export const model = new RideViewModel();
  * Window to edit properties of the currently selected ride.
  */
 export const rideWindow = window({
-	title: model._title,
+	title: "Edit ride",
```
Comment (Basssiiie) [inline src/ui/rideWindow.ts:25]: Does it not work with the ride's title in there?

File: src/ui/rideWindow.ts [hunk 4]
```diff
@@ -21,10 +22,10 @@ export const model = new RideViewModel();
...
 	position: "center",
-	width: { value: 233, min: 185, max: 250 },
-	height: 252,
+	width: 233, minWidth: 185, maxWidth: 250,
```
Comment (Basssiiie) [inline src/ui/rideWindow.ts:27]: This seems to be reverting to changes from an earlier version as well?

File: src/ui/rideWindow.ts [hunk 5]
```diff
@@ -115,6 +116,28 @@ export const rideWindow = window({
 				})
 			]
 		}),
+		groupbox({
+			text: "Ride name",
+			tooltip: "Edit the name of the ride",
+			content: [
+				label({
+					text: model._title,
+					height: 14
```
Comment (Basssiiie) [inline src/ui/rideWindow.ts:125]: 14 is the default height of a label, so this line can be omitted.

File: src/ui/rideWindow.ts [hunk 6]
```diff
@@ -115,6 +116,28 @@ export const rideWindow = window({
...
+				}),
+				button({
+					text: "Rename ride",
+					height: buttonSize,
+					onClick: () => {
+						ui.showTextInput({
+							title: "Ride/attraction name",
+							initialValue: model._title.get(),
+							description: "Enter new name for this ride/attraction:",							
+							callback: (input) => context.executeAction("ridesetname", {ride: model._ride.get()?._id, name: input})
```
Comment (Basssiiie) [inline src/ui/rideWindow.ts:135]: Shouldn't we update the store as well once the `ridesetname` action is successfully completed?

**Review by Basssiiie — CHANGES_REQUESTED** [review]
Hey thank you for your PR! I think it's a nice addition. 😄 I do have some nitpicks before merging though.

File: package-lock.json (hunk 1 above)
Comment (Manticore-007) [reply package-lock.json]: Thanks for the feedback, reverted this mistake.

File: src/ui/mainWindow.ts (hunk 2 above)
Comment (Manticore-007) [reply src/ui/mainWindow.ts]: Corrected it, I didn't update flexui before continuing in my own fork.

File: src/ui/rideWindow.ts (hunk 3 above)
Comment (Manticore-007) [reply src/ui/rideWindow.ts]: Reverted it back. Something didn't work out when I was trying out stuff and forget to revert it to the original.

File: src/ui/rideWindow.ts (hunk 4 above)
Comment (Manticore-007) [reply src/ui/rideWindow.ts]: Corrected it, I didn't update flexui before continuing in my own fork as well.

File: src/ui/rideWindow.ts (hunk 5 above)
Comment (Manticore-007) [reply src/ui/rideWindow.ts]: removed the line.

File: src/ui/rideWindow.ts (hunk 6 above)
Comment (Manticore-007) [reply src/ui/rideWindow.ts]: added it.

**Comment by Manticore-007** [comment]
> I do have some nitpicks before merging though.

Not nitpicking. I need the feedback to do a PR correctly.

File: src/ui/rideWindow.ts [hunk 7]
```diff
@@ -115,6 +116,30 @@ export const rideWindow = window({
...
+					onClick: () => {
+						ui.showTextInput({
+							title: "Ride/attraction name",
+							initialValue: model._title.get(),
+							description: "Enter new name for this ride/attraction:",							
+							callback: (input) => {
+								context.executeAction("ridesetname", {ride: model._ride.get()?._id, name: input});
+								model._title.set(input);
+							}
+						})
```
Comment (Basssiiie) [inline src/ui/rideWindow.ts:138]:
```suggestion
						});
```
Linting was giving a warning for this. (CI should fail for this though, weird that it doesn't. 🤔)

**Review by Basssiiie — COMMENTED** [review]
Last 2 points. The rest looks and works great! 😄

File: src/ui/rideWindow.ts [hunk 8]
```diff
@@ -115,6 +116,30 @@ export const rideWindow = window({
...
+					text: "Rename ride",
+					height: buttonSize,
+					onClick: () => {
+						ui.showTextInput({
+							title: "Ride/attraction name",
+							initialValue: model._title.get(),
+							description: "Enter new name for this ride/attraction:",							
+							callback: (input) => {
+								context.executeAction("ridesetname", {ride: model._ride.get()?._id, name: input});
+								model._title.set(input);
```
[the remark this answers is not in the document]
Comment (Manticore-007) [reply src/ui/rideWindow.ts]: Added it to the PR!

File: src/ui/rideWindow.ts (hunk 7 above)
Comment (Manticore-007) [reply src/ui/rideWindow.ts]: corrected it :-)

**Review by Basssiiie — APPROVED** [review]
Thanks!
